### PR TITLE
Replaced repeated code with view component for results table 

### PR DESCRIPTION
**Pull request by Ted-U**
in orgs, schemes, users and logs

### CODE REVIEW

File: app/components/view_result_component.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class ViewResultComponent < ViewComponent::Base
```
Comment (baarkerlounger) [inline app/components/view_result_component.rb]: Maybe `SearchResultCaptionComponent`?

**Comment by baarkerlounger** [comment]
Be good to see some unit tests for the component and looks like it needs linting (`bundle exec rake lint`) but otherwise :+1:

File: app/components/view_result_component.rb (hunk 1 above)
Comment (Ted-U) [reply app/components/view_result_component.rb]: I have updated the component name as suggested

**Comment by Ted-U** [comment]
> Be good to see some unit tests for the component and looks like it needs linting (`bundle exec rake lint`) but otherwise 👍

Linting is done

File: app/components/search_result_caption_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+  <span class="govuk-!-margin-right-4">
+    <% if searched.present? %>
+      <strong><%= count %></strong> <%= item_label %> found matching ‘<%= searched %>’ of <strong><%= total_count %></strong> total <%= item %>. <%= govuk_link_to("Clear search", request.path) %>
+    <% else %>
+      <strong><%= count %></strong> total <%= item %>.
+    <% end %>
```
Comment (baarkerlounger) [inline app/components/search_result_caption_component.html.erb:6]: Looks like the `title` is missing here

Comment (baarkerlounger) [inline app/components/search_result_caption_component.html.erb:6]: Looks like the `title` is missing here

**Comment by Ted-U** [comment]
> Be good to see some unit tests for the component and looks like it needs linting (`bundle exec rake lint`) but otherwise 👍

Unit tests now added

File: app/components/search_result_caption_component.html.erb (hunk 2 above)
Comment (Ted-U) [reply app/components/search_result_caption_component.html.erb:6]: Title has been replaced with item but I can change that for clarity if needed?

**Review by baarkerlounger — APPROVED** [review]

**Review by moarpheus — APPROVED** [review]

**Pull request merged**